src/status_im2/navigation/core.cljs

```diff
@@ -51,6 +52,11 @@
      (when-not @state/curr-modal
        (reset! state/pushed-screen-id view-id)))))
 
+(navigation/reg-component-did-disappear-listener
```

Comment:
unfortunately, this won't work, because `did-disappear-listener` is very late, this is our pain point, we need to listen for swipe on ios, same how we do with back button on android

Reply:
so it will be possible to swipe quick and open chat, and `did-disappear-listener` will happen and chat screen will be broken

Reply:
> so it will be possible to swipe quick and open chat, and `did-disappear-listener` will happen and chat screen will be broken

Yes, this can be problematic. @churik please can you check if this is reproducible for you?

Reply:
it was reported before so its possible

Reply:
apparently this is[ known issue](https://stackoverflow.com/questions/63444571/how-to-detect-swipe-back-event-in-iphone-react-native), and solution is to rely on navigation listeners (which doesn't work for us).
Another solution I can try to is to call :chat/close on unmount

Reply:
yup it works, will push changes